File: Numbers/NextPrime/prime.js
const BigNumber = require('bignumber.js')

exports.nextPrime = function (n) {
  let i = n
  while (i) {
    i = i.plus(1)
    if (isPrime(i)) {
      return i
    }
  }
}

function isPrime(n) {
  let c = {}
  c[new BigNumber(0)] = new BigNumber(1)
  for (let i = new BigNumber(0); i.lt(n); i = i.plus(1)) {
    c[i.plus(1)] = new BigNumber(1)
    for (let j = i; j.gt(0); j = j.minus(1)) {
      c[j] = c[j.minus(1)].minus(c[j])
    }
    c[new BigNumber(0)] = c[new BigNumber(0)].negated()
  }

  c[new BigNumber(0)] = c[new BigNumber(0)].plus(1)
  c[new BigNumber(n)] = c[new BigNumber(n)].minus(1)

  for (let i = n; i.gte(0); i = i.minus(1)) {
    if (!c[i].modulo(n).isEqualTo(0)) {
      return false
    }
  }

  return true
}

File: python/Numbers/NextPrime/prime.py
def is_prime(n):
    c = {}
    c[0] = 1
    for i in range(0, n):
        c[i + 1] = 1
        for j in range(i, 0, -1):
            c[j] = c[j - 1] - c[j]
        c[0] = -c[0]
    
    c[0] = c[0] + 1
    c[n] = c[n] - 1

    for i in range(n, 0, -1):
        if(c[i] % n != 0):
            return False
    
    return True

def next_prime(n):
    i = n
    while True:
        i += 1
        if is_prime(i):
            return i